LeaderboardBackend.Test/Leaderboards.cs

```diff
@@ -35,24 +35,15 @@ public async Task OneTimeSetUp()
     }
 
     [OneTimeTearDown]
-    public void OneTimeTearDown()
-    {
-        _factory.Dispose();
-    }
+    public void OneTimeTearDown() => _factory.Dispose();
 
     [Test]
-    public void GetLeaderboard_NotFound()
-    {
-        RequestFailureException e = Assert.ThrowsAsync<RequestFailureException>(
-            async () =>
-                await _apiClient.Get<LeaderboardViewModel>(
+    public void GetLeaderboard_NotFound() => _apiClient.Invoking(a => a.Get<LeaderboardViewModel>(
                     $"/api/leaderboards/{long.MaxValue}",
                     new()
-                )
-        )!;
-
-        Assert.AreEqual(HttpStatusCode.NotFound, e.Response.StatusCode);
-    }
+                )).Should()
+                .ThrowAsync<RequestFailureException>()
+                .Where(e => e.Response.StatusCode == HttpStatusCode.NotFound);
```

Comment:
This needs to be awaited.

Reply:
The chained call to `Where` returns a `Task` that needs to be awaited. See https://fluentassertions.com/exceptions/.

The correct code is something like
```cs
await FluentActions.Awaiting(async () => await _apiClient.Get<LeaderboardViewModel>(
    $"/api/leaderboards/{long.MaxValue}",
    new()
)).Should().ThrowAsync<RequestFailureException>()
.Where(e => e.Response.StatusCode == HttpStatusCode.NotFound);
```

Alternatively, use a `Func`:
```cs
Func<Task<LeaderboardViewModel>> act = async () => await _apiClient.Get<LeaderboardViewModel>(
    $"/api/leaderboards/{long.MaxValue}",
    new()
);

await act.Should().ThrowAsync<RequestFailureException>()
.Where(e => e.Response.StatusCode == HttpStatusCode.NotFound);
```

Reply:
Oh whoops. Didn't realise that my test was passing for me because the result wasn't even resolved.